sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpandExec.scala

```diff
@@ -163,7 +163,7 @@ class GpuExpandIterator(
         val (cv, nullColumnReused) = expr.columnarEval(cb) match {
           case null => getOrCreateNullCV(sparkType)
           case other =>
-            (GpuExpressionsUtils.resolveColumnVector(other, cb.numRows, sparkType), false)
+            (GpuExpressionsUtils.resolveColumnVector(other, cb.numRows), false)
```

Comment:
Does the return value "other" of expr.columnarEval() have the same type as the expr.dataType?

Reply:
As long as the expression itself is correctly programmed, sparkType is the result type of the "expr".

Reply:
Yes. You can find it here
https://github.com/NVIDIA/spark-rapids/blob/branch-22.02/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpandExec.scala#L162